vcf/utils.py

```diff
@@ -37,19 +37,19 @@ def walk_together(*readers, **kwargs):
         next_idx_to_k = dict(
             (i, get_key(r)) for i, r in enumerate(nexts) if r is not None)
         keys_with_prev_contig = [
-            k for k in next_idx_to_k.values() if k[0] == min_k[0]]
+            k for k in list(next_idx_to_k.values()) if k[0] == min_k[0]]
```

Comment:
I don't think the call to `list` adds anything here.
```suggestion
            k for k in next_idx_to_k.values() if k[0] == min_k[0]]
```